prototype/mc_shortest_path/arduino_mc_shortest_path/arduino_mc_shortest_path.ino

```diff
@@ -111,39 +111,87 @@ void getPossibleActions(Board* board, Action* possibleActions) {
   }
 }
 
-int getNumPossibleActions() { return BOARD_SIZE - 1; }
+bool isDone(Board* board) {
+  return (board->values[BOARD_SIZE - 1] == -1 ||
+          board->values[BOARD_SIZE - 1] == -2);
+}
+
+int getNumPossibleActions(Board* board) {
+  if (isDone(board)) {
+    return 0;
+  }
+  int nActions = 0;
+  for (int i = 0; i < BOARD_SIZE; ++i) {
+    if (board->values[i] != -2 && board->values[i] != -1) {
+      nActions++;
+    }
+  }
+  return nActions;
+}
 
 int getBoardSize() { return BOARD_SIZE; }
 
-bool isDone(Board* board) {
-  if (board->values[BOARD_SIZE - 1] == -1)
+bool isStuck(Board* board) {
```

Comment:
It would be much more efficient in this method to do a switch case on current pos and return base on your conditions.

```
bool isStuck(const Board* board)
{
  uint8_t currentPos = findCurrentPosition(board);
  switch(currentPos)
  {
    case 0:
      return (..)
    case 1:
      return (...)
  }
}
```